Templates/templates/CSharp/Requests/EntityCollectionRequest.cs.tt

```diff
@@ -50,18 +50,18 @@ namespace <#=@namespace#>
             {
                 if (response.AdditionalData != null)
                 {
-                    object nextPageLink;
-                    response.AdditionalData.TryGetValue("@odata.nextLink", out nextPageLink);
-
-                    var nextPageLinkString = nextPageLink as string;
-
-                    if (!string.IsNullOrEmpty(nextPageLinkString))
+                    if(response.AdditionalData.TryGetValue("@odata.nextLink", out var nextPageLink))
                     {
-                        response.Value.InitializeNextPageRequest(
-                            this.Client,
-                            nextPageLinkString);
+                        // Ensure it is a non empty JsonElement string
+                        if (nextPageLink is System.Text.Json.JsonElement element
+                            && element.ValueKind == System.Text.Json.JsonValueKind.String
+                            && !string.IsNullOrEmpty(element.ToString()))
```

Comment:
shouldn't this be get string instead? https://docs.microsoft.com/en-us/dotnet/api/system.text.json.jsonelement.getstring?view=net-5.0#System_Text_Json_JsonElement_GetString

Reply:
That is valid. Especially since we have ensured that the valuekind is String.
Will action and resolve.

Reply:
Resolved via https://github.com/microsoftgraph/MSGraph-SDK-Code-Generator/pull/512/commits/7fa45931929c82c4696e864ad0b714110f104e2c